### PR TITLE
[SHLWAPI] Fix SHLoadIndirectString executing code when loading a string

### PR DESCRIPTION
**Pull request by learn-more**

### CODE REVIEW

File: modules/rostests/apitests/shlwapi/SHLoadIndirectString.c [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+    HRSRC hRsrc = FindResourceW(GetModuleHandleW(NULL), ResourceName, (LPCWSTR)10);
+    ok(!!hRsrc, "Unable to find %s\n", wine_dbgstr_w(ResourceName));
+    if (!hRsrc)
+        return FALSE;
+
+    hGlobal = LoadResource(GetModuleHandleW(NULL), hRsrc);
+    Size = SizeofResource(GetModuleHandleW(NULL), hRsrc);
+    pData = LockResource(hGlobal);
+
+    ok(Size && !!pData, "Unable to load %s\n", wine_dbgstr_w(ResourceName));
```
Comment (KyleKatarn31) [inline modules/rostests/apitests/shlwapi/SHLoadIndirectString.c:82]: Why !!pData instead of pData ?

**Comment by KyleKatarn31** [comment]
> 
> 
> _No description provided._

Maybe to consider a PR description here ?

File: modules/rostests/apitests/shlwapi/shlwapi_resource_dll/shlwapi_resource_dll.c [hunk 2]
```diff
@@ -0,0 +1,28 @@
+#define WIN32_NO_STATUS
+#define _INC_WINDOWS
+#define COM_NO_WINDOWS_H
+
+#include <windows.h>
+
+
```
Comment (HBelusca) [inline modules/rostests/apitests/shlwapi/shlwapi_resource_dll/shlwapi_resource_dll.c:7]:
```suggestion
```

File: modules/rostests/apitests/shlwapi/SHLoadIndirectString.c [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+            dwRet = WaitForSingleObject(hEvent, 1);
+            ok_hex(dwRet, WAIT_OBJECT_0);
+            FreeLibrary(mod);
+        }
+    }
+    CloseHandle(hEvent);
+}
+
+
+BOOL extract_resource(const WCHAR* Filename, LPCWSTR ResourceName)
```
Comment (katahiromz) [inline modules/rostests/apitests/shlwapi/SHLoadIndirectString.c:66]: Make the function static

File: modules/rostests/apitests/shlwapi/SHLoadIndirectString.c [hunk 4]
```diff
@@ -0,0 +1,133 @@
...
+    // Show that it is not signaled
+    dwRet = WaitForSingleObject(hEvent, 1);
+    ok_hex(dwRet, WAIT_TIMEOUT);
+
+    // Ensure the module is not loaded yet...
+    HMODULE mod = GetModuleHandleW(DllFile);
+    if (mod != NULL)
+    {
+        skip("%S loaded, cannot continue\n", DllFile);
+        return;
```
Comment (katahiromz) [inline modules/rostests/apitests/shlwapi/SHLoadIndirectString.c:31]: Close event

**Comment by learn-more** [comment]
> The fix can be sent to wine (since they have the same code):
> https://github.com/wine-mirror/wine/blob/6d801377055911d914226a3c6af8d8637a63fa13/dlls/shlwapi/string.c#L2842

You volunteer?

File: modules/rostests/apitests/shlwapi/SHLoadIndirectString.c (hunk 3 above)
Comment (learn-more) [reply modules/rostests/apitests/shlwapi/SHLoadIndirectString.c:66]: No, other apitests can share the function if it's not static.

File: modules/rostests/apitests/shlwapi/SHLoadIndirectString.c (hunk 1 above)
Comment (JoachimHenze) [reply modules/rostests/apitests/shlwapi/SHLoadIndirectString.c:83]: This is a proper way for a C++ developer to cast something to bool without receiving a compiler warning. This is ok.

**Comment by KyleKatarn31** [comment]
> > The fix can be sent to wine (since they have the same code):
> > https://github.com/wine-mirror/wine/blob/6d801377055911d914226a3c6af8d8637a63fa13/dlls/shlwapi/string.c#L2842
> 
> You volunteer?

Is it something still to be done ?

**Pull request merged**